doc/sphinx/language/core/records.rst

```diff
@@ -15,188 +15,226 @@ synonym :cmd:`Structure`.
+   .. insertprodn record_definition field_type
 
    .. prodn::
       record_definition ::= {? > } @ident_decl {* @binder } {? : @sort } {? := {? @ident } %{ {*; @record_field } {? ; } %} {? as @ident } }
-      record_field ::= {* #[ {*, @attribute } ] } @name {? @field_body } {? %| @natural } {? @decl_notations }
-      field_body ::= {* @binder } @of_type
+      record_field ::= {* #[ {*, @attribute } ] } @name {? @field_type } {? %| @natural }
+      field_type ::= {* @binder } @of_type
       | {* @binder } @of_type := @term
       | {* @binder } := @term
 
-   Defines the record named
-   :n:`@ident_decl`. For each field, :n:`@name` is the name of the
-   field. Fields can be :gdef:`abstract <abstract field>` as in
-   :n:`@name {* @binder } @of_type` or :gdef:`manifest <manifest
-   field>`, that is explicitly specified, as in :n:`@name {* @binder } @of_type := @term`.
-   Fields have a type given by :n:`@of_type` (when absent, as in :n:`@name {* @binder } := @term`,
-   the type is inferred automatically).  In manifest
-   fields, :n:`@term` is called the *body* of the field. The body
-   and type of a field can depend on previous fields, so the order
-   of fields in the definition may matter.
-
-   To a record is associated a constructor. The name of this
-   constructor is provided by the occurrence of :n:`@ident` after
-   :n:`:=`. When omitted, the constructor name defaults to
-   :n:`Build_@ident` where :n:`@ident` is the record name.
-
-   A record type belongs to a sort which is given by :token:`sort`. If
-   omitted, the default sort is Type.
-
-   A sequence of :n:`@binder` may be applied to the record as a whole to declare
-   the *parameters* of the record. They may also be applied to individual fields.
-   For example, :n:`{+ @binder } : @of_type` is
-   equivalent to :n:`forall {+ @binder } , @of_type` and
-   similarly for the other forms of :n:`@field_body`.
-
-   When possible, the :cmd:`Record` command associates to each
-   field of the record a projection for accessing the instance
-   of the field in an object of type :token:`ident`.  These
-   projections are given the names of the corresponding fields. If a
-   field is named `_` then no projection is built for it.
-
-   If given, the :n:`as @ident` part in the syntax of records
-   specifies the name to use for referring to the argument corresponding to the
-   record in the type of projections.
-
-   If :n:`{? > }` is provided, the constructor is automatically declared as
-   a coercion from the class of the last field type to the record name
-   (this may fail if the uniform inheritance condition is not
-   satisfied). See :ref:`coercions`.
-
-   Notations can be attached to fields using the :n:`@decl_notations` annotation.
+   Defines a record type and creates projections for each field that has a name other
+   than `_`. The field body and type can depend on previous
+   fields, so the order of fields in the definition may matter.
+
+   :n:`{? > }`
```

Comment:
Our convention is to describe items in the order they appear in the syntax so it's easy to find their descriptions rather than in order of importance.

Reply:
OK, if this is the convention.